php-frontend/src/main/java/org/sonar/plugins/php/api/tree/declaration/ClassPropertyDeclarationTree.java

```diff
@@ -45,6 +45,8 @@
 @Beta
 public interface ClassPropertyDeclarationTree extends ClassMemberTree {
 
+  List<AttributeTree> attributes();
```

Comment:
We lose some information by exposing `attributes` rather than `attributeGroups`. We lose the fact that some attributes are grouped together and we don't have access to tokens such as `#[`. This kind of information is very rarely used, but I believe this AST API should expose expose all the details of the syntax and not a more useful interpretation of the syntax.

I would not have this `attributes` method. I would not provide such "helper" method in the public API, at least for now.

However, I believe it could be useful to create a new interface which is extended by `ClassPropertyDeclarationTree`, `ClassTree`... to represent the fact that they can have attributes:
```
interface HasAttributes {
  List<AttributeGroupTree> attributeGroups();
}
```